frontend/src/console/components/Hero/Hero.tsx

```diff
@@ -82,35 +81,51 @@ export class HeroComponent extends React.PureComponent<Props, State> {
 
     this.state = {
       title: '',
-      subtitle: ''
+      subtitle: '',
+      renderBool: true
     };
   }
 
   public componentDidMount() {
     this.checkAndUpdateState();
   }
 
-  public componentDidUpdate() {
+  public componentDidUpdate(prevProps: Props) {
     this.checkAndUpdateState();
+    if (
+      this.activeInstanceDataExists() &&
+      this.staticContentOverridesExists() &&
+      prevProps.activeInstance.data &&
+      this.props.activeInstance.data &&
+      prevProps.activeInstance.data.draftStaticContentOverrides !==
+        this.props.activeInstance.data.draftStaticContentOverrides
```

Comment:
There are redundant checks here. `this.staticContentOverridesExists()` already calls `activeInstanceDataExists()`, which checks `this.props.activeInstance.data`.
Could we simplify this by doing something like this?
```ts
public componentDidUpdate(prevProps: Props) {
  this.checkAndUpdateState();
  if (
    this.staticContentOverridesExists() &&
    this.staticContentOverridesExists(prevProps) &&
    prevProps.activeInstance.data!.draftStaticContentOverrides !==
      this.props.activeInstance.data!.draftStaticContentOverrides
  ) {
    this.checkNewAndUpdateState();
  }
}

...

private activeInstanceDataExists = (props = this.props) => {
  return props.activeInstance && props.activeInstance.data;
};

private themeConfigExists = (props = this.props) => {
  return (
    this.activeInstanceDataExists(props) &&
    props.activeInstance.data!.draftThemeConfig
  );
};

private staticContentOverridesExists = (props = this.props) => {
  return (
    this.activeInstanceDataExists(props) &&
    props.activeInstance.data!.draftStaticContentOverrides
  );
};
```

Reply:
Yes, this makes sense, I have updated it, now every method taking props as an argument and performing the instructions based on the props received

Reply:
@rusrushal13, it would be more convenient to have the default value for `props` for these checks. That's why I put `(props = this.props)` above. This way we won't need to pass `this.props` all the time, which will make it easier to read and minimize the changes of the unrelated lines in this PR.

Reply:
Thanks, @Agrendalath, having default arguments surely helps, I have updated it in my latest commit :smiley: